apps/admin/components/Post/PostUploadPage/RestaurantSearchSection/index.tsx

```diff
@@ -0,0 +1,99 @@
+import { Badge, Box, Input, VStack } from "@chakra-ui/react";
+import { css } from "@emotion/react";
+import { width100 } from "@toss/emotion-utils";
+import { useCallback, useEffect, useState } from "react";
+import { Text } from "~/components/Common";
+import { useInput } from "~/hooks";
+import { useRestaurantSetter } from "~/recoil/atoms/search";
+
+const RestaurantSearchSection = () => {
+  const { props: keyword, debouncedValue } = useInput({
+    initialValue: "",
+    useDebounce: true,
+    debounceTimeout: 300
+  });
+
+  const setRestaurant = useRestaurantSetter();
+
+  const [result, setResult] = useState<kakao.maps.services.PlacesSearchResult>(
+    []
+  );
+
+  useEffect(() => {
+    const ps = new kakao.maps.services.Places();
+
+    if (debouncedValue)
+      ps.keywordSearch(
+        debouncedValue,
+        (data, status) => {
+          switch (status) {
+            case kakao.maps.services.Status.OK:
+              console.log("성공");
+              setResult(data);
+              break;
+            case kakao.maps.services.Status.ZERO_RESULT:
+              console.log("검색 결과가 없습니다.");
+              break;
+            case kakao.maps.services.Status.ERROR:
+            case null:
+              console.error("오류발생");
+              break;
+            default:
+          }
+        },
+        {
+          x: 37.566826,
+          y: 126.9786567
+        }
+      );
+  }, [debouncedValue]);
+
+  const onItemClickHandler = useCallback(
+    (item: kakao.maps.services.PlacesSearchResultItem) => {
+      setRestaurant({
+        address_name: item.address_name,
+        category_group_name: item.category_group_name,
+        id: item.id,
+        place_name: item.place_name,
+        road_address_name: item.road_address_name,
+        x: item.x,
+        y: item.y
+      });
```

Comment:
```suggestion
    setRestaurant(item);
```

이렇게 변경해도 되지 않나요? 전 잘 동작하긴 합니다!

Reply:
오 잘 작동하면 좋네요 

item의 타입이 현재 아래와같습니다.


```tsx
  /**
     * 장소 ID
     */
    id: string;

    /**
     * 장소명, 업체명
     */
    place_name: string;

    /**
     * 카테고리 이름
     * 예) 음식점 > 치킨
     */
    category_name: string;

    /**
     * 중요 카테고리만 그룹핑한 카테고리 그룹 코드
     * 예) FD6
     */
    category_group_code: CategoryGroupCode;

    /**
     * 중요 카테고리만 그룹핑한 카테고리 그룹명
     * 예) 음식점
     */
    category_group_name: string;

    /**
     * 전화번호
     */
    phone: string;

    /**
     * 전체 지번 주소
     */
    address_name: string;

    /**
     * 전체 도로명 주소
     */
    road_address_name: string;

    /**
     * X 좌표값 혹은 longitude
     */
    x: string;

    /**
     * Y 좌표값 혹은 latitude
     */
    y: string;

    /**
     * 장소 상세페이지 URL
     */
    place_url: string;

    /**
     * 중심좌표까지의 거리(x,y 파라미터를 준 경우에만 존재). 단위 meter
     */
    distance: string;
```

그러면 item을 왕창 넣어버리면 필요없는거까지 들어가지 않나요?

차라리 코드가 좀 길어지더라도, 내가 사용할것들을 명확히 명시함으로써 다른 개발자가 추후 합류해도 이해하기 빠를거같아요 
어떤게 더 나으신가요? 전 둘다 괜찮아보이긴합니다.

Reply:
오 전 찬혁님이 그냥 원래 썼던 코드가 더 낫네요. 왕왕 들어가는 것 보다는 내가 필요한 값들만 쏙쏙 넣는 게 맞는 것 같습니다!